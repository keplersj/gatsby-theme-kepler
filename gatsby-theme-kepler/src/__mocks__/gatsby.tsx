import * as React from "react";
const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      /* eslint-disable no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      /* eslint-enable no-unused-vars */
      to,
      ...rest
      // eslint-disable-next-line jsx-a11y/anchor-has-content
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
