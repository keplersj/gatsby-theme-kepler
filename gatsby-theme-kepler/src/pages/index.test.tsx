import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Page from "./index";
import {
  KeplerAvatarComponentQuery,
  KeplerBannerBackgroundData,
  KeplerNavbarData,
  KeplerBaseLayoutData,
  KeplerIndexPageData
} from "../__mockData__";
import deepMerge from "deepmerge";

beforeEach((): void => {
  (useStaticQuery as jest.Mock).mockImplementation((): object =>
    deepMerge.all([
      KeplerBannerBackgroundData,
      KeplerAvatarComponentQuery,
      KeplerNavbarData,
      KeplerBaseLayoutData
    ])
  );
});

describe("Index Page", (): void => {
  it("renders correctly", (): void => {
    const tree = renderer
      .create(<Page location={{ pathname: "" }} data={KeplerIndexPageData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
