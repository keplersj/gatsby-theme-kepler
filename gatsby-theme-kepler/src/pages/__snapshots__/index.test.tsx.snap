// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders correctly 1`] = `
Array [
  <script
    type="application/ld+json"
  >
    Object {
    "@context": "https://schema.org",
    "@type": "BreadcrumbList",
    "itemListElement": Array [
      Object {
        "@type": "ListItem",
        "item": Object {
          "@id": "https://site.dev/",
          "@type": "WebPage",
          "name": "Kepler Sticka-Jones",
        },
        "position": 1,
      },
    ],
  }
  </script>,
  <script
    type="application/ld+json"
  >
    Object {
    "@context": "https://schema.org",
    "@type": "WebSite",
    "about": Object {
      "@type": "Person",
      "description": "This is an example of a personal web site.",
      "email": "test@example.dev",
      "image": "",
      "name": "Example Site",
      "sameAs": Array [
        "https://github.dev/test",
        "https://twitter.dev/test",
        "https://linkedin.dev/in/test",
        "https://keybase.dev/test",
        "https://instagram.dev/test",
        "https://ü§∑üèº‚Äç‚ôÇÔ∏è.dev/test",
      ],
      "url": "https://site.dev",
    },
    "description": "This is an example of a personal web site.",
    "name": "Example Site",
    "url": "https://site.dev",
  }
  </script>,
  .emotion-0 {
  position: relative;
}

.emotion-0::before,
.emotion-0::after {
  position: absolute;
  top: 0;
  left: 0;
  display: block;
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  -webkit-transition: opacity 0.5s ease 0.25s;
  transition: opacity 0.5s ease 0.25s;
  content: "";
}

@media screen and (prefers-reduced-motion:reduce) {
  .emotion-0::before,
  .emotion-0::after {
    -webkit-transition: none;
    transition: none;
  }
}

.emotion-0::after {
  z-index: -100;
  opacity: 0;
}

.emotion-0::before {
  z-index: -101;
  opacity: 1;
  background-image: url("data+image/svg+xml");
}

@media screen and (prefers-color-scheme:dark) {
  .emotion-0::before {
    background-image: url("data+image/svg+xml");
  }
}

.emotion-1 {
  min-height: 66vh;
  -webkit-backdrop-filter: blur(20px);
  backdrop-filter: blur(20px);
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3rem 1.5rem;
  color: #fff;
}

.emotion-3 {
  text-align: center!important;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 0 auto;
  position: relative;
  width: auto;
}

@media screen and (min-width: 1024px) {
  .emotion-3 {
    max-width: 960px;
  }
}

@media screen and (min-width: 1216px) {
  .emotion-3 {
    max-width: 1152px;
  }
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -0.75rem;
  margin-right: -0.75rem;
  margin-top: -0.75rem;
}

.emotion-4:last-child {
  margin-bottom: -0.75rem;
}

.emotion-5 {
  margin-left: auto;
  margin-right: auto;
  display: block;
  padding: 0.75rem;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 66.6666%;
}

.emotion-6 {
  margin-bottom: 10px;
  border-radius: 140px;
}

.emotion-7 {
  font-size: 2.5rem;
  font-weight: 600;
  line-height: 1.125;
  margin-bottom: 1.5rem;
}

.emotion-8 {
  margin-left: auto;
  margin-right: auto;
}

.emotion-9 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 2.25em;
  padding-top: calc(0.375em - 1px);
  padding-right: 0.75em;
  padding-bottom: calc(0.375em - 1px);
  padding-left: 0.75em;
  color: #000;
  font-size: 1.5rem;
  line-height: 1.5;
  white-space: nowrap;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: top;
  border: 1px solid transparent;
  border-color: transparent;
  border-width: 1px;
  border-radius: 4px;
  box-shadow: none;
  cursor: pointer;
  -moz-appearance: none;
  -webkit-appearance: none;
  color: #fff;
}

.emotion-9:hover,
.emotion-9:focus {
  background-color: #5f5f5f;
}

.emotion-9:active {
  background-color: #373737;
}

@media (prefers-color-scheme: dark) {
  .emotion-9 {
    color: #fff;
  }

  .emotion-9:hover,
  .emotion-9:focus {
    background-color: #a0a0a0;
  }

  .emotion-9:active {
    background-color: #c8c8c8;
  }
}

<div
    className="emotion-0"
    data-testid="BackgroundImageContainerElement"
  >
    <noscript
      className="lc82qt"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "noscript{z-index:-100;opacity:1;background-image:url(\\"/static/banner_example/3/banner.jpg\\"); @media screen and (prefers-color-scheme: dark){background-image:url(\\"/static/banner_dark_example/3/banner_dark.jpg\\");};;}",
          }
        }
      />
    </noscript>
    <section
      className="emotion-1"
    >
      <div
        className="emotion-2"
      >
        <div
          className="emotion-3"
        >
          <div
            className="emotion-4"
          >
            <div
              className="emotion-5"
            >
              <div
                aria-label="Avatar"
                role="img"
              >
                <div
                  className="emotion-6 gatsby-image-wrapper"
                  style={
                    Object {
                      "display": "inline-block",
                      "height": 200,
                      "overflow": "hidden",
                      "position": "relative",
                      "width": 200,
                    }
                  }
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    src="data+image/svg+xml"
                    style={
                      Object {
                        "height": "100%",
                        "left": 0,
                        "objectFit": "cover",
                        "objectPosition": "center",
                        "opacity": 1,
                        "position": "absolute",
                        "top": 0,
                        "transitionDelay": "500ms",
                        "width": "100%",
                      }
                    }
                  />
                  <picture>
                    <source
                      srcSet="/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x"
                    />
                    <img
                      alt="Avatar"
                      height={200}
                      loading="lazy"
                      onLoad={[Function]}
                      src="/static/example/1/avatar.jpg"
                      srcSet="/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x"
                      style={
                        Object {
                          "height": "100%",
                          "left": 0,
                          "objectFit": "cover",
                          "objectPosition": "center",
                          "opacity": 0,
                          "position": "absolute",
                          "top": 0,
                          "transition": "opacity 500ms",
                          "width": "100%",
                        }
                      }
                      width={200}
                    />
                  </picture>
                  <noscript
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<picture><source srcset=\\"/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x\\" /><img loading=\\"lazy\\" width=\\"200\\" height=\\"200\\" srcset=\\"/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x\\" src=\\"/static/example/1/avatar.jpg\\" alt=\\"Avatar\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                      }
                    }
                  />
                </div>
              </div>
              <h1
                className="emotion-7"
              >
                Example Site
              </h1>
              <div
                className="emotion-8"
              >
                <a
                  className="emotion-9"
                  href="/blog"
                >
                  Blog
                </a>
                <a
                  className="emotion-9"
                  href="/portfolio"
                >
                  Portfolio
                </a>
                <a
                  className="emotion-9"
                  href="/about"
                >
                  About
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  .emotion-0 {
  padding: 1em;
}

@media screen and (min-width: 512px) {
  .emotion-0 {
    padding: 1em 2.5em;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    padding: 1em 5em;
  }
}

<section
    className="emotion-0"
  >
    <div>
      <h2>
        Biography
      </h2>
      <section
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>I am a computer programmer and college student based out of Salt Lake City, with experience in entrepreneurship, student leadership, and open source software development.</p>",
          }
        }
      />
    </div>
    <a
      href="/about"
    >
      Read More...
    </a>
  </section>,
  .emotion-0 {
  padding: 1em;
}

@media screen and (min-width: 512px) {
  .emotion-0 {
    padding: 1em 2.5em;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    padding: 1em 5em;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: scroll;
}

.emotion-2 {
  margin: 1em;
  background-color: rgba(var(--starstuff-card-default-background-color), 0.5);
  border-radius: 10px;
  box-shadow: 0.5em 0.5em 0.75em rgba(var(--starstuff-card-default-shadow-color), 0.3);
  -webkit-transition: box-shadow 1s;
  transition: box-shadow 1s;
  --starstuff-card-default-background-color: 245,245,245;
  --starstuff-card-default-shadow-color: 10,10,10;
  min-width: 20em;
  max-width: 25vw;
}

.emotion-2:active,
.emotion-2:focus,
.emotion-2:focus-within,
.emotion-2:hover {
  box-shadow: 0.8em 0.8em 1em rgba(var(--starstuff-card-default-shadow-color), 0.4);
}

@media screen and (prefers-color-scheme: dark) {
  .emotion-2 {
    --starstuff-card-default-background-color: 55,55,55;
  }
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.5em;
}

.emotion-4 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: flex-end;
  align-self: flex-end;
}

.emotion-6 .gatsby-image-wrapper {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
}

<section
    className="emotion-0"
  >
    <script
      type="application/ld+json"
    >
      Object {
      "@context": "https://schema.org",
      "@id": "https://site.dev/blog",
      "@type": "Blog",
      "blogPost": Array [
        Object {
          "@id": "https://site.dev/blog/01-post-without-image",
          "@type": "BlogPosting",
        },
        Object {
          "@id": "https://site.dev/blog/02-post-with-image",
          "@type": "BlogPosting",
        },
      ],
      "mainEntityOfPage": "https://site.dev/blog",
      "url": "https://site.dev/blog",
    }
    </script>
    <h2>
      Blog Posts
    </h2>
    <div
      className="emotion-1"
    >
      <div
        className="emotion-2"
      >
        <script
          type="application/ld+json"
        >
          Object {
          "@context": "https://schema.org",
          "@id": "https://site.dev/blog/01-post-without-image",
          "@type": "BlogPosting",
          "datePublished": "2018-12-18",
          "headline": "Test Post (without Image)",
          "mainEntityOfPage": "https://site.dev/blog/01-post-without-image",
          "name": "Test Post (without Image)",
          "url": "https://site.dev/blog/01-post-without-image",
        }
        </script>
        <div
          className="emotion-3"
        >
          <a
            href="/blog/01-post-without-image"
          >
            <h3>
              Test Post (without Image)
            </h3>
          </a>
          <time
            dateTime="2018-12-18"
          >
            Dec 18, 2018
          </time>
          <p>
            This is an example of a blog post without a featured image.
          </p>
          <a
            className="emotion-4"
            href="/blog/01-post-without-image"
          >
            Read More...
          </a>
        </div>
      </div>
      <div
        className="emotion-2"
      >
        <script
          type="application/ld+json"
        >
          Object {
          "@context": "https://schema.org",
          "@id": "https://site.dev/blog/02-post-with-image",
          "@type": "BlogPosting",
          "datePublished": "2018-12-18",
          "headline": "Test Post (with Image)",
          "image": Object {
            "@id": "https://site.dev/static/image-1/image.jpeg",
            "@type": "ImageObject",
          },
          "mainEntityOfPage": "https://site.dev/blog/02-post-with-image",
          "name": "Test Post (with Image)",
          "url": "https://site.dev/blog/02-post-with-image",
        }
        </script>
        <a
          aria-hidden={true}
          href="/blog/02-post-with-image"
        >
          <figure
            className="emotion-6"
          >
            <script
              type="application/ld+json"
            >
              Object {
              "@context": "https://schema.org",
              "@id": "https://site.dev/static/image-1/image.jpeg",
              "@type": "ImageObject",
              "contentUrl": "/static/image-1/image.jpeg",
              "representativeOfPage": false,
              "url": "/static/image-1/image.jpeg",
            }
            </script>
            <div
              className=" gatsby-image-wrapper"
              style={
                Object {
                  "maxHeight": null,
                  "maxWidth": null,
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                aria-hidden={true}
                style={
                  Object {
                    "paddingBottom": "100%",
                    "width": "100%",
                  }
                }
              />
              <img
                alt=""
                aria-hidden={true}
                src="data+image/svg+xml"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transitionDelay": "500ms",
                    "width": "100%",
                  }
                }
              />
              <picture>
                <source
                  sizes="(min-width 100px)"
                  srcSet="/static/image-1/image.jpeg 1x"
                />
                <img
                  alt=""
                  loading="lazy"
                  onLoad={[Function]}
                  sizes="(min-width 100px)"
                  src="/static/image-1/image.jpeg"
                  srcSet="/static/image-1/image.jpeg 1x"
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 500ms",
                      "width": "100%",
                    }
                  }
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><source srcset=\\"/static/image-1/image.jpeg 1x\\" sizes=\\"(min-width 100px)\\" /><img loading=\\"lazy\\" sizes=\\"(min-width 100px)\\" srcset=\\"/static/image-1/image.jpeg 1x\\" src=\\"/static/image-1/image.jpeg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
          </figure>
        </a>
        <div
          className="emotion-3"
        >
          <a
            href="/blog/02-post-with-image"
          >
            <h3>
              Test Post (with Image)
            </h3>
          </a>
          <time
            dateTime="2018-12-18"
          >
            Dec 18, 2019
          </time>
          <p>
            This is an example of a blog post with a featured image.
          </p>
          <a
            className="emotion-4"
            href="/blog/02-post-with-image"
          >
            Read More...
          </a>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`Index Page renders correctly 2`] = `
<html
  lang="en"
>
  <head>
    <link
      data-rh={true}
      href="https://site.dev/"
      rel="canonical"
    />
    <meta
      content=""
      data-rh={true}
      name="description"
    />
    <meta
      content="Kepler Sticka-Jones"
      data-rh={true}
      property="og:site_name"
    />
    <meta
      content="Kepler Sticka-Jones"
      data-rh={true}
      property="og:title"
    />
    <meta
      content="https://site.dev/"
      data-rh={true}
      property="og:url"
    />
    <meta
      content=""
      data-rh={true}
      property="og:description"
    />
    <meta
      content="summary"
      data-rh={true}
      name="twitter:card"
    />
    <meta
      content="@test"
      data-rh={true}
      name="twitter:site"
    />
    <meta
      content="@test"
      data-rh={true}
      name="twitter:creator"
    />
    <title
      data-rh={true}
    >
      Kepler Sticka-Jones
    </title>
  </head>
  <body />
</html>
`;
